### PR TITLE
Command system rewrite

### PR DESCRIPTION
**Pull request by Snowiiii**

### CODE REVIEW

File: pumpkin/src/commands/cmd_help.rs
Comment (lukas0008) [inline pumpkin/src/commands/cmd_help.rs:1]: Don't keep commented code, i mean this is what git is for, if you want an old version, use the history

**Review by lukas0008 — COMMENTED** [review]
Overall very nice

**Comment by Bryntet** [comment]
I understood it as if this would allow async within commands, but it doesn't seem like the case? Maybe I'm missing something.

File: pumpkin/src/commands/cmd_help.rs
Comment (Snowiiii) [reply pumpkin/src/commands/cmd_help.rs:1]: Yeah. I just knew that i would reimplement it soon and im too lazy to search trough commits

Comment (lukas0008) [reply pumpkin/src/commands/cmd_help.rs:1]: > Don't keep commented code, i mean this is what git is for, if you want an old version, use the history

Resolved by 870059a31762289e774199cb61880c6fce1b77cb

**Comment by Snowiiii** [comment]
> I understood it as if this would allow async within commands, but it doesn't seem like the case? Maybe I'm missing something.

I said async is now way easier to implement. Be now can actually implement it without big problems if there where no async clouseres thing. Using the old approche async would be impossible

**Comment by Bryntet** [comment]
Anyways, other than me still wanting async in commands, this looks good to me.

**Comment by Snowiiii** [comment]
> Anyways, other than me still wanting async in commands, this looks good to me.

No worries. It will come

**Pull request merged**